### PR TITLE
Fix strlogicalcmp

### PR DESCRIPTION
**Pull request by janisozaur**
We recently had reports about invalid sorting of such items

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Your new rewrite appears to have done away with case insensitivity. Could you restore this?

**Comment by janisozaur** [comment]
Can you provide the suggestion to tests?

**Comment by IntelOrca** [comment]
`EXPECT_EQ(strlogicalcmp("foo13", "FOO13"), 0);`

**Comment by AaronVanGeffen** [comment]
When you said "reports about invalid sorting of such items", were you thinking of any issues in particular? I recall there were issues with lists not being ordered as expected, but they turned out to be using alphabetical order rather than logical order.

**Comment by janisozaur** [comment]
I think there was a report on reddit, not here.

**Comment by AaronVanGeffen** [comment]
I wish you'd posted it in the OP, as it's hard to see exactly what this PR is trying to fix now. Combined with the regression mentioned above, how would you like to proceed?

File: src/openrct2/util/Util.cpp [hunk 1]
```diff
@@ -323,37 +323,32 @@ int32_t strcicmp(char const* a, char const* b)
...
-                return na - nb;
-            }
+        if (*s2 == '\0')
+            return *s1 != '\0';
+        else if (*s1 == '\0')
+            return -1;
+        else if (!(isdigit(*s1) && isdigit(*s2)))
+        {
+            if (tolower(*s1) != tolower(*s2))
+                return (int)*s1 - (int)*s2;
```
Comment (IntelOrca) [inline src/openrct2/util/Util.cpp:337]:
```suggestion
                return tolower(*s1) - tolower(*s2);
```

Comment (IntelOrca) [reply src/openrct2/util/Util.cpp:337]: EXPECT_LT(strlogicalcmp("A", "b"), 0);
EXPECT_LT(strlogicalcmp("a", "B"), 0);
EXPECT_GT(strlogicalcmp("B", "a"), 0);
EXPECT_GT(strlogicalcmp("b", "A"), 0);

**Comment by IntelOrca** [comment]
What is this actually fixing again?

**Pull request merged**